cli/internal/cmd/iamcreateaws.go

```diff
@@ -65,22 +72,44 @@ func iamCreateAWS(cmd *cobra.Command, spinner spinnerInterf, creator iamCreator)
 
 	// Creation.
 	spinner.Start("Creating", false)
+
+	var conf *config.Config
+	if awsFlags.generateConfig {
+		conf, err = createConfig(cmd, fileHandler, cloudprovider.AWS, awsFlags.configPath)
+		if err != nil {
+			return err
+		}
+	}
+
 	iamFile, err := creator.Create(cmd.Context(), cloudprovider.AWS, &cloudcmd.IAMConfig{
 		AWS: cloudcmd.AWSIAMConfig{
 			Region: awsFlags.region,
 			Prefix: awsFlags.prefix,
 		},
 	})
+
 	spinner.Stop()
 	if err != nil {
 		return err
 	}
 
-	cmd.Printf("region:\t%s\n", awsFlags.region)
-	cmd.Printf("zone:\t%s\n", awsFlags.zone)
-	cmd.Printf("iamProfileControlPlane:\t%s\n", iamFile.AWSOutput.ControlPlaneInstanceProfile)
-	cmd.Printf("iamProfileWorkerNodes:\t%s\n", iamFile.AWSOutput.WorkerNodeInstanceProfile)
-	cmd.Println("Your IAM configuration was created successfully. Please fill the above values into your configuration file.")
+	if awsFlags.generateConfig {
+		conf.Provider.AWS.Region = awsFlags.region
+		conf.Provider.AWS.Zone = awsFlags.zone
+		conf.Provider.AWS.IAMProfileControlPlane = iamFile.AWSOutput.ControlPlaneInstanceProfile
+		conf.Provider.AWS.IAMProfileWorkerNodes = iamFile.AWSOutput.WorkerNodeInstanceProfile
+
+		if err := fileHandler.WriteYAML(awsFlags.configPath, conf, file.OptOverwrite); err != nil {
+			return err
+		}
+		cmd.Printf("Your IAM configuration was created and filled into %s successfully.\n", awsFlags.configPath)
+	} else {
```

Comment:
Use early return instread of else.
```suggestion
        return nil
	}
```